internal/sim/sim.go

```diff
@@ -348,16 +349,18 @@ func (s *Simulator) step() {
 		runOp = iota
 		deliverCall
 		deliverReply
+		deliverCallError
+		deliverReplyError
 	)
 	var candidates []int
 	if s.numOps < s.opts.NumOps {
 		candidates = append(candidates, runOp)
 	}
 	if len(s.calls) > 0 {
-		candidates = append(candidates, deliverCall)
+		candidates = append(candidates, deliverCall, deliverCallError)
```

Comment:
Will this lead to shallow executions, where too many calls/replies end up failing?

Fine to leave as-is for now.

Reply:
Yeah, I think you're 100% right. I implemented the simplest thing for now, but in the future, I'll have to improve things. I don't have anything super concrete in mind, but one idea for example, is that we should probably simulate executions without any injected failures first and only inject errors if we can't find a failing execution.